### PR TITLE
Amazon Bedrock Agents for Finanacial Analyst Code Samples

### PR DESCRIPTION
**Pull request by ucegbe**
*Description of changes:* I added the folder for the amazon bedrock agents examples for financial investment research that Sovik and team worked on 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/files/FSI-PortfolioTool-BedrockAgent-9a4cdb9f-14b9-4658-9720-537735bf2b1c.zip
Comment (mttanke) [inline agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/files/FSI-PortfolioTool-BedrockAgent-9a4cdb9f-14b9-4658-9720-537735bf2b1c.zip:1]: we should make those available as files and zip them + package for S3 with code

File: agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/files/FSI-SentimentDetecttion-BedrockAgent-6aebee60-d2df-464c-b652-b93e2aef7b3a.zip
Comment (mttanke) [inline agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/files/FSI-SentimentDetecttion-BedrockAgent-6aebee60-d2df-464c-b652-b93e2aef7b3a.zip:1]: we should make those available as files and zip them + package for S3 with code

File: agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/files/FSI-StockQuery-BedrockAgent-4bed2482-a84f-4d9b-b53c-37220fd4b9bb.zip
Comment (mttanke) [inline agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/files/FSI-StockQuery-BedrockAgent-4bed2482-a84f-4d9b-b53c-37220fd4b9bb.zip:1]: we should make those available as files and zip them + package for S3 with code

File: agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/files/FSI-TextractAsyncInvocationFunction-759363ea-7b4b-411c-a96f-b137a595387b.zip
Comment (mttanke) [inline agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/files/FSI-TextractAsyncInvocationFunction-759363ea-7b4b-411c-a96f-b137a595387b.zip:1]: we should make those available as files and zip them + package for S3 with code

File: agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/files/FSI-TextractProcessingFunction-b6054c5e-0286-4cfb-a2f3-55d408a9972c.zip
Comment (mttanke) [inline agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/files/FSI-TextractProcessingFunction-b6054c5e-0286-4cfb-a2f3-55d408a9972c.zip:1]: we should make those available as files and zip them + package for S3 with code

File: agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/files/FSI-Transcribe-66a4860d-390a-4af5-aa1b-05fb469be7ac.zip
Comment (mttanke) [inline agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/files/FSI-Transcribe-66a4860d-390a-4af5-aa1b-05fb469be7ac.zip:1]: we should make those available as files and zip them + package for S3 with code

File: agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/images/kb6.png
Comment (mttanke) [inline agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/images/kb6.png:1]: remove KB ID

File: agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/images/kb7.png
Comment (mttanke) [inline agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/images/kb7.png:1]: remove the S3 url

File: agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/images/kendra_sync.png
Comment (mttanke) [inline agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/images/kendra_sync.png:1]: what is kendra being used for?

File: agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/images/stack_resources.png
Comment (mttanke) [inline agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/images/stack_resources.png:1]: remove physical id

**Review by mttanke — CHANGES_REQUESTED** [review]
We need to make the code more automated and remove manual steps plus too permissive roles

File: agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/README.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+## Implementation Steps
+
+## Prerequisites
+
+This solution uses [Lambda functions](https://aws.amazon.com/lambda/), which are serverless, event-driven compute services that runs applications. The Python code plus other dependencies for the applications are packaged as zip files, stored in *files* in this repo. We need to add them to an S3 bucket in your account in order to set up the Lambda functions.
+
+Please use the us-west-2 region.
+
+First, make an S3 bucket. Go to S3 page in AWS, click "Create bucket". Then enter a bucket name, which should be universally unique. Take a note of the name, because we will need it in following sections. Leave the rest as default, and click "Create bucket" at the bottom of the page.
```
[the remark this answers is not in the document]
Comment (oashton) [reply agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/README.md:17]: before proceeding with the changes:
1. Is it ok if we upload the files (i.e.: .zip lambda layers, pdf, json and csv) to the “aws-blogs-artifacts-public” bucket and then do the copy of file to the S3 account?, please see [this](https://github.com/aws-samples/agentsforbedrock-retailagent/blob/main/template.yml) as a reference.
2. About the lambda source code .zip files, ¿is it ok if we put the source code inline in the CF template?

File: agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/images/kb6.png
Comment (ucegbe) [reply agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/images/kb6.png:1]: done

File: agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/images/kb7.png
Comment (ucegbe) [reply agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/images/kb7.png:1]: done

File: agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/images/stack_resources.png
Comment (ucegbe) [reply agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/images/stack_resources.png:1]: done

File: agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/images/kendra_sync.png
Comment (ucegbe) [reply agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/images/kendra_sync.png:1]: original implementation with langchain used Kendra, I have removed the pictures as the current implementation uses KB

File: agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/README.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
+# Generative AI and Multi-Modal Agents in AWS: The Key to Unlocking New Value in Financial Markets
+
+This file walks you through how to set up the infrastructure and applications, and run the code to create a multi-modal agent using Amazon Bedrock. The blog post provides a detailed discussion of this solution. This Solution uses Full-Access policies for certain services involved in the architecture and may not meet the security standards of certain business units, however, this solution should be used to show the art-of-possible and access policy should be scoped down for any other use.
```
[the remark this answers is not in the document]
Comment (ucegbe) [reply agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/README.md:3]: Yes

File: agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/README.md [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+<img src="images/select_files.png" width="680"/>
+
+<img src="images/files_uploaded.png" width="680"/>
+
+\
+This Solution use Anthropic Claude 3 model in Bedrock. Make sure you have access to the model in Bedrock.
+Navigate to the Bedrock console and click on **Model Access** to the left tab. Select the **Manage model access** button to the top-right.
+
+<img src="images/bedrock-access.JPG" width="680"/>
```
[the remark this answers is not in the document]
Comment (ucegbe) [reply agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/README.md]: fixed

File: agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/README.md [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+\
+Leave the rest as default. Check the acknowledgement box on the last page and submit it. 
+
+It takes a few minutes to create. Once it is created, you can check the generated resources by clicking on Stacks -> Stack Name. Then click "Resources". The example below shows that the AudioTranscriptsSourceBucketResource is an S3 bucket with the bucket name "test-cf-stack-audiotranscriptssourcebucketresourc-1kg41ts9dy7hk".
+
+<img src="images/stack_resources.png" width="680"/>
+
+## Generate text files from audio and pdf files
+
+[AWS CloudShell](https://aws.amazon.com/cloudshell/) allos you to explore and manage AWS resources from a terminal in your browser
```
[the remark this answers is not in the document]
Comment (ucegbe) [reply agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/README.md:77]: fixed

File: agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/README.md [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+2. For the code in the [Lambda functions](https://aws.amazon.com/lambda/), it's important to know what the agent send to the Lambda function and what should be response. For the request you can check the general format in this [link](https://docs.aws.amazon.com/bedrock/latest/userguide/agents-lambda.html#agents-lambda-input), in the *parameters* or *requestBody* fields you can get the data necessary to execute the logic depending of your needs. For the response you can check the general format in this [link](https://docs.aws.amazon.com/bedrock/latest/userguide/agents-lambda.html#agents-lambda-response), it's important to generate in the code a response object that comply with structure defined in the corresponding OpenAPI Schema.
+
+## Security
+
+See [CONTRIBUTING](CONTRIBUTING.md#security-issue-notifications) for more information.
```
[the remark this answers is not in the document]
Comment (ucegbe) [reply agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/README.md:218]: done

File: agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/AWS Pricing Calculator Estimation.pdf
[the remark this answers is not in the document]
Comment (ucegbe) [reply agents-for-bedrock/use-case-examples/ai-powered-assistant-for-investment-research/AWS Pricing Calculator Estimation.pdf:1]: I have removed it as the solution will be run in workshop accounts and cost is very dynamic as model pricing can change

**Review by ucegbe — COMMENTED** [review]
I have made the non-cloudformation changes requested.

**Review by mttanke — APPROVED** [review]

**Pull request merged**